node/handlers/dagstats.go

```diff
@@ -0,0 +1,103 @@
+package handlers
+
+import (
+	"context"
+
+	coreiface "github.com/ipfs/boxo/coreiface"
+	"github.com/ipfs/boxo/ipld/merkledag"
+	"github.com/ipfs/boxo/ipld/merkledag/traverse"
+	"github.com/ipfs/go-cid"
+	"go.uber.org/atomic"
+)
+
+const (
+	DagStatsCtxKey = "ctx_dagstats"
+)
+
+func getDagStatsFromCtx(ctx context.Context) *DagStats {
+	ds, ok := ctx.Value(DagStatsCtxKey).(*DagStats)
+	if !ok {
+		return nil
+	}
+	return ds
+}
+
+type DagStats struct {
+	DeduplicatedSize      *atomic.Int64
```

Comment:
DeduplicatedSize and DeduplicatedNumBlocks may not be as important when we shard the DAG across different nodes. Once sharded, these two metrics may no longer provide accurate insights